### DISCUSSION TITLE
Make it possible to get all data?

### DISCUSSION THREAD

**Discussion by Joey-1970** [theengs/decoder#108] [General]
Hello all!

I search for a BLE to MQTT converter on a raspberry pi, but the sensors are not in the supported device list. I want to get the data from two MOPEKA gas sensor (pro and standard) but I need only raw-data. The target is a home automation software called ip-symcon. So the question is: Is it possible to get a new option like "-a" for all raw data?

Joachim

**Comment by 1technophile** [comment]
Hello,

You can use TheengsGateway, we've just added the [option](https://github.com/theengs/gateway/commit/30995680df2421afbcecd980cad027afc224ee05) to publish all sensors `-pa` 
https://github.com/theengs/gateway
Also if you are interested we can integrate the Mopeka into the decoder.

Florian

**Comment by Joey-1970** [comment]
Hello Florian,

that sounds great! How do I get the new version? What must I do to get it? Is there something like "upgrade" or "reinstall"? I try "pip3 install --upgrade TheengsGateway" but I can't see the new option in -h help...

I can help to integrate the senosors, but I need your help.
I try [here](https://github.com/Joey-1970/Mopeka/blob/main/MopekaGasSensor/module.php) to decode the protocol, but until now I can't verified it. The pro-sensor maybe will work in your app, but the standard-sensor I actually don't understand to decode myself...

Joachim

**Reply by 1technophile** [reply]
Hello Joachim,

I'm going to release it today or tomorrow. I will let you know.

**Reply by 1technophile** [reply]
New version [released](https://github.com/theengs/gateway/releases), you can upgrade this way:
`pip3 install TheengsGateway --upgrade`

**Comment by Joey-1970** [comment]
Hello Florian,

**many thanks for this!**

I now can see the data from the MOPEKA-sensors and I will try to decode alle of the features (temperature, gas level, etc). I will come back and we can define a "device" for your app.

One further question: What is the best way to get your app to start with boot?

Joachim

**Reply by 1technophile** [reply]
Hello,

Cool!
For the app to start at boot , you can take a look to the solution below:
https://github.com/theengs/gateway/discussions/5
